blocks/panel-color/index.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import { omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { PanelColor } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import ColorPalette from '../color-palette';
+import withColorContext from '../with-color-context';
+
+function ConditionalPanelColor( { title, hasColorsToChoose, value, ...props } ) {
+	if ( ! hasColorsToChoose ) {
+		return null;
+	}
+	return (
+		<PanelColor title={ title } colorValue={ value }>
+			<ColorPalette
+				value={ value }
+				{ ...omit( props, [ 'disableCustomColors', 'colors' ] ) }
+			/>
+		</PanelColor>
+	);
+}
+
+export default withColorContext( ConditionalPanelColor );
```

Comment:
To avoid `if ( ! hasColorsToChoose ) {` check it can be expressed as:
```js
export default compose(
    withColorContext,
    ifCondition( ( { hasColorsToChoose } ) => hasColorsToChoose ),
)( PanelColor );
```

Aside: `ifCondition` could support string to pass a prop name :)